.github/workflows/lint.yml

```
@@ -8,32 +8,24 @@ on:
   pull_request:
 
 jobs:
-  backend:
+  lint:
+    strategy:
+      matrix:
+        workspace:
+        - backend-771cdf
+        - client-66101c
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
       with:
+        fetch-depth: 0
         submodules: true
-    - uses: actions/setup-node@v3
+    - uses: actions/setup-node@v3.2.0
       with:
         node-version: 18.x
-        cache: 'yarn'
-        cache-dependency-path: |
-          packages/backend/yarn.lock
-    - run: yarn install
-    - run: yarn --cwd ./packages/backend lint
-
-  client:
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v2
-      with:
-        submodules: true
-    - uses: actions/setup-node@v3
-      with:
-        node-version: 18.x
-        cache: 'yarn'
-        cache-dependency-path: |
-          packages/client/yarn.lock
-    - run: yarn install
-    - run: yarn --cwd ./packages/client lint
+    - run: corepack enable
+    - run: yarn set version berry
+    - run: yarn workspaces foreach install --immutable

```

```suggestion
    - run: yarn install --immutable
```
If workspaces are enabled, it should be sufficient to run install only on the root.